src/rez/pip.py

```diff
@@ -134,11 +132,19 @@ def find_python_in_context(context):
     # Create a copy of the context with systems paths removed, so we don't
     # accidentally find a system python install.
     #
+    success = (context.status == ResolverStatus.solved)
+    if not success:
+        context.print_info(buf=sys.stderr)
+        return None
+
     context = context.copy()
     context.append_sys_path = False  # #826
 
     python_package = context.get_resolved_package("python")
-    assert python_package
+    if not python_package:
```

Comment:
Will this condition ever be true? If we resolved python and the solve was successful, `context.get_resolved_package("python")` should always return a python package no?